pandas/core/groupby/ops.py

```diff
@@ -637,43 +681,23 @@ def _cython_operation(
             if not is_complex_dtype(dtype):
                 values = ensure_float64(values)
 
-        arity = self._cython_arity.get(how, 1)
         ngroups = self.ngroups
+        comp_ids, _, _ = self.group_info
 
         assert axis == 1
         values = values.T
-        if how == "ohlc":
-            out_shape = (ngroups, 4)
-        elif arity > 1:
-            raise NotImplementedError(
-                "arity of more than 1 is not supported for the 'how' argument"
-            )
-        elif kind == "transform":
-            out_shape = values.shape
-        else:
-            out_shape = (ngroups,) + values.shape[1:]
-
-        func, values = self._get_cython_func_and_vals(kind, how, values, is_numeric)
-
-        if how == "rank":
-            out_dtype = "float"
-        else:
-            if is_numeric:
-                out_dtype = f"{values.dtype.kind}{values.dtype.itemsize}"
-            else:
-                out_dtype = "object"
 
-        codes, _, _ = self.group_info
+        out_shape = cy_op.get_output_shape(ngroups, values)
```

Comment:
future, would just have a single method here where you pass `ngroup`, `values`, *may* be able to determine `is_numeric` inside the WrappedCythonOps (instead of here), and just return the items you need, though you can take it a step further and have a `WrappedAggregateOp` and `WrappedTransformOp`, something like

```
op = WrappedCythonOps(kind, how, values.....) # other args
# op is WrappedTransform / WrappedAggregate
result = op.get_result()
```